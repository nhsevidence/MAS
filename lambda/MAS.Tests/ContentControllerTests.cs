using System.IO;
using System.Threading.Tasks;

using Xunit;
using Amazon.Lambda.TestUtilities;
using Amazon.Lambda.APIGatewayEvents;

using Newtonsoft.Json;

namespace MAS.Tests
{
    public class ContentControllerTests
    {
        [Fact]
        public async Task TestPut()
        {
            var lambdaFunction = new LambdaEntryPoint();

            var requestStr = File.ReadAllText("./SampleRequests/ContentController-Put.json");
            var request = JsonConvert.DeserializeObject<APIGatewayProxyRequest>(requestStr);
            var context = new TestLambdaContext();
            var response = await lambdaFunction.FunctionHandlerAsync(request, context);

            Assert.Equal(200, response.StatusCode);
        }
    }
}
