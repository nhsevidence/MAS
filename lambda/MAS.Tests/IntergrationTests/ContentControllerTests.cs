using System.Threading.Tasks;
using Xunit;
using Newtonsoft.Json;
using MAS.Tests.Infrastructure;
using System.Net.Http;
using System.Text;
using System.Net;
using Shouldly;

namespace MAS.Tests.IntergrationTests
{
    public class ContentControllerTests : TestBase
    {
        [Fact]
        public async Task Put()
        {
            //Arrange
            var content = new StringContent(JsonConvert.SerializeObject(""), Encoding.UTF8, "application/json");

            //Act
            var response = await _client.PutAsync("/api/content/1234", content);

            // Assert
            response.StatusCode.ShouldBe(HttpStatusCode.OK);
        }
    }
}
