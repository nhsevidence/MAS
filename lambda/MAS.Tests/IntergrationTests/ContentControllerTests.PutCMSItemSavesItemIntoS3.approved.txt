Title: Some title
Short Summary: Wonder drug
Source: The Journal of Medicine
Evidence Type: Evidence Type
UKMI Comment: UKMI Comment
Resource Links: <p><a title="Link 1" href="items/5de65fe432281d43fbfcd15a">Link 1</a></p>
<p><a title="sadada" href="items/5de65fe432281d43fbfcd15a">Link 2</a></p>